compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2102,6 +2102,15 @@ public void visit(BLangXMLNavigationAccess xmlNavigation) {
         if (xmlNavigation.childIndex != null) {
             analyzeExpr(xmlNavigation.childIndex);
         }
+        validateMethodInvocationsInXMLNavigationExpression(xmlNavigation);
+    }
+
+    private void validateMethodInvocationsInXMLNavigationExpression(BLangXMLNavigationAccess expression) {
+        if (!expression.methodInvocationAnalyzed && expression.parent.getKind() == NodeKind.INVOCATION) {
+            dlog.error(((BLangInvocation) expression.parent).name.pos,
```

Comment:
Can't we use
```suggestion
            dlog.error(((BLangInvocation) expression.parent).pos,
```
instead?

Reply:
Wanted to get the error highlighting just for method name + paranthesis, but just realized that paranthesis will not be included in the position. Will changed to suggested.